### PR TITLE
Add server half of message passing code.

### PR DESCRIPTION
**Pull request by djpetti**
Again, the server class does exactly what it sounds like: it implements a server that listens on a port.

Interestingly, all the actual listening, as well as the handling of clients, is done in separate threads, so this should allow us to use a single-threaded implementation in situations where we were previously using a multi-threaded implementation.

This might not be entirely complete; I think there are still a few tests that I want to write. However, I think it's good enough to review/use.

### CODE REVIEW

File: common/message_passing/client.h [hunk 1]
```diff
@@ -23,9 +24,9 @@
...
 
 /**
- * @brief Handles sending messages to another node.
+ * @brief Handles connecting to another node and sending/receiving messages.
  */
-class Client {
+class Client : public Node {
  public:
   /**
    * @param thread_pool Thread pool to use internally for managing associated
```
Comment (jake-chandler) [inline common/message_passing/client.h:32]: I'm not sure what you mean by "managing associated"

**Review by jake-chandler — APPROVED** [review]
Looks straightforward to use. Thanks!

**Review by Nickster258 — APPROVED** [review]
Hard to critique this, looks really clean!

File: common/message_passing/client.h (hunk 1 above)
Comment (djpetti) [reply common/message_passing/client.h:32]: Each `Client` instance requires a `SenderTask` and a `ReceiverTask`, which are the "associated tasks" to be managed.

**Pull request merged**